src/bluez/peripheral.rs

```diff
@@ -60,33 +73,57 @@ impl Peripheral {
         }
     }
 
-    fn characteristic_info(&self, characteristic: &Characteristic) -> Result<CharacteristicInfo> {
-        let services = self.services.lock().unwrap();
+    fn get_characteristic<'a>(
+        &self,
```

Comment:
It looks like you're no longer using this `self` parameter, so might as well remove it.